src/CommonDBTM.php

```diff
@@ -5938,11 +5938,23 @@ protected function update1NTableData(
         $commondb_relation = new $commondb_relation();
 
         // Compute which item is item_1 and item_2
-        if (static::class == $commondb_relation::$itemtype_1) {
+        if (
+            static::class == $commondb_relation::$itemtype_1
+            || (
+                preg_match('/^itemtype/', $commondb_relation::$itemtype_1) === 1
+                && preg_match('/^itemtype/', $commondb_relation::$itemtype_2) === 0
+            )
```

Comment:
This should be done after the `static::class == $commondb_relation::$itemtype_2` case.

Reply:
Done.